### PR TITLE
codeView: Add hover loops for flip-to-hack button

### PR DESCRIPTION
**Pull request by ptomato**
Same scheme as in the codeview to gracefully handle a sound being
stopped before the UUID arrives.

https://phabricator.endlessm.com/T24523

### CODE REVIEW

File: js/ui/codeView.js [hunk 1]
```diff
@@ -275,13 +275,42 @@ var WindowTrackingButton = new Lang.Class({
         this.child.flipped = value == STATE_TOOLBOX;
     },
 
+    _startHoverSound(flipped) {
+        this._hoverSoundID = 'pending';
+        const id = flipped ? 'flip-inverse' : 'flip';
+        SoundServer.getDefault().playAsync(`shell/tracking-button/${id}/hover`)
+        .then(uuid => {
+            print('received uuid');
```
Comment (cosimoc) [inline js/ui/codeView.js]: Remove this

**Comment by cfoch** [comment]
I was implementing this, too. I had the problem that if I passed the mouse very quick over the arrow button, a loop kept playing even when the mouse was out of the button. The same problem occurs here.

**Comment by cfoch** [comment]
It seems that the calls to the hover signal handler is faster than the response of the uuid from the server.

**Comment by cosimoc** [comment]
@cfergeau That's what the `pending`/`cancel` status is supposed to catch in this PR as far as I can see.

**Comment by ptomato** [comment]
It's certainly possible that it doesn't yet work right!

**Comment by cosimoc** [comment]
@ptomato I can reproduce the issue that @cfoch mentioned here easily by hovering on and off the flip button in quick succession.
I think what is happening is that the callback returns the UUID after the second hover already changed the state back to `pending`, so the first sound never gets cancelled.

**Pull request merged**